src/test/java/core/basesyntax/FileServiceImplTest.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax;
+
+import core.basesyntax.service.FileService;
+import core.basesyntax.service.implementions.FileServiceImpl;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FileServiceImplTest {
+    private static final String FILE_NAME_FROM = "src/test/java/resources/inputCorrect.csv";
+    private static final String FILE_NAME_TO = "src/test/java/resources/reportPerfect.csv";
+    private static final String FILE_NAME_WRONG = "src/test/java/resources/wrong.csv";
+    private static FileService fileService;
+
+    @BeforeClass
+    public static void beforeClass() {
+        fileService = new FileServiceImpl();
+    }
+
+    @Test
+    public void readDataFromFile_isOk() {
+        List<String> actualResult = fileService.readDataFromFile(FILE_NAME_FROM);
+        List<String> expectedResult = List.of("type,fruit,quantity", "b,banana,20",
+                            "b,apple,100", "s,banana,100", "p,banana,13", "r,apple,10");
+        Assert.assertEquals(expectedResult, actualResult);
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void readDataFromFile_NonExistentFile_NotOk() {
+        fileService.readDataFromFile(FILE_NAME_WRONG);
+    }
+
+    @Test
+    public void writeToFile_isOk() {
+        String expectedReport = "fruit,quantity\n" + "apple,90\n" + "banana,152\n";
```

Comment:
If you use the Mac symbol "\n" can't be work. Better use System.lineSeparator()

Reply:
done